src/platform/user/profile/utilities/index.js

```diff
@@ -207,26 +212,5 @@ export const showAddressValidationModal = suggestedAddresses => {
     return false;
   }
 
-  if (
-    suggestedAddresses.length === 1 &&
-    (addressMetaData.deliveryPointValidation === BAD_UNIT_NUMBER ||
-      addressMetaData.deliveryPointValidation === MISSING_UNIT_NUMBER)
-  ) {
-    return true;
-  }
-
-  if (suggestedAddresses.length > 1) {
-    return true;
-  }
-
-  if (
-    suggestedAddresses.length === 1 &&
-    addressMetaData.deliveryPointValidation !== CONFIRMED
-  ) {
-    return true;
-  }
-
-  return (
-    suggestedAddresses.length === 1 && addressMetaData.confidenceScore < 80
-  );
+  return true;
```

Comment:
All the existing unit tests passed with this simplification. And besides this is almost certainly the correct logic.